scripts/dndrks/less_concepts.lua

```diff
@@ -0,0 +1,803 @@
+    out7 = (rule_as_binary[3] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods7) then
+    out7 = (rule_as_binary[2] << 1) & 2
+  elseif compare (seed_pack7, neighborhoods8) then
+    out7 = (rule_as_binary[1] << 1) & 2
+  else out7 = (0 << 1) & 2
+  end
+end
+
+function com8()
+if compare (seed_pack8,neighborhoods1) then
+    out8 = rule_as_binary[8] & 1
+  elseif compare (seed_pack8, neighborhoods2) then
+    out8 = rule_as_binary[7] & 1
+  elseif compare (seed_pack8, neighborhoods3) then
+    out8 = rule_as_binary[6] & 1
+  elseif compare (seed_pack8, neighborhoods4) then
+    out8 = rule_as_binary[5] & 1
+  elseif compare (seed_pack8, neighborhoods5) then
+    out8 = rule_as_binary[4] & 1
+  elseif compare (seed_pack8, neighborhoods6) then
+    out8 = rule_as_binary[3] & 1
+  elseif compare (seed_pack8, neighborhoods7) then
+    out8 = rule_as_binary[2] & 1
+  elseif compare (seed_pack8, neighborhoods8) then
+    out8 = rule_as_binary[1] & 1
+  else out8 = 0 & 1
+  end
+end
+
+com1()
+com2()
+com3()
+com4()
+com5()
+com6()
+com7()
+com8()
+
+next_seed = out1+out2+out3+out4+out5+out6+out7+out8
+
+end
+
+function notes_off_v1()
+  for i=1,#active_notes_v1 do
+    m.note_off(active_notes_v1[i],0,ch_1)
+  end
+  active_notes_v1 = {}
+end
+
+function notes_off_v2()
+  for i=1,#active_notes_v2 do
+    m.note_off(active_notes_v2[i],0,ch_2)
+  end
+  active_notes_v2 = {}
+end
+
+-- if user-defined bit in the binary version of a seed equals 1, then note event [aka, bit-wise gating]
+function iterate()
+  notes_off_v1()
+  notes_off_v2()
+  seed = next_seed
+    bang()
+    scale(new_low,new_high,seed)
+    if seed_as_binary[v1_bit] == 1 then
+      engine.noteOn(1,midi_to_hz((notes[coll][scaled])+(48+(v1_octave * 12)+semi)),127)
+      m.note_on((notes[coll][scaled])+(36+(v1_octave*12)+semi),127,ch_1)
+      table.insert(active_notes_v1,(notes[coll][scaled])+(36+(v1_octave*12)+semi))
+    end
+    if seed_as_binary[v2_bit] == 1 then
+      engine.noteOn(2,midi_to_hz((notes[coll][scaled])+(48+(v2_octave * 12)+semi)),127)
+      m.note_on((notes[coll][scaled])+(36+(v2_octave*12)+semi),127,ch_2)
+      table.insert(active_notes_v2,(notes[coll][scaled])+(36+(v2_octave*12)+semi))
+    end
+    redraw()
+    grid_redraw()
+end
+
+-- convert midi note to hz for Passersby engine
+function midi_to_hz(note)
+  return (440 / 32) * (2 ^ ((note - 9) / 12))
+end
+
+-- allow user to define the MIDI channel voice 1 sends on
+function midi_vox_1(channel)
+  ch_1 = channel
+end
+
+-- allow user to define the MIDI channel voice 2 sends on
+function midi_vox_2(channel)
+  ch_2 = channel
+end
+
+-- allow user to define the transposition of voice 1 and voice 2, simultaneous changes to MIDI and Passersby engine
+function transpose(semitone)
+  semi = semitone
+end
+
+-- everything that happens when the script is first loaded
+function init()
+  math.randomseed(os.time())
+  math.random(); math.random(); math.random()
```

Comment:
why generate 3 random numbers that aren't used?

Reply:
based on comments from folks in Lua forums, the first 'random' number is not pure random -- it was advised to trigger a few at the start, to overcome. is this true?